### PR TITLE
Add redux store

### PR DESCRIPTION
**Pull request by YuCJ**
Base on #22 

[story-3] 身為 @twreporter/redux 的 package consumer，我希望不用再自己組 store 和重寫 mapDispatchToAction，可以直接用打包好的 store 和 async method
[TASK](https://kanbanflow.com/t/ef1bKfVA)

### CODE REVIEW

**Comment by nickhsine** [comment]
Please merge this one https://github.com/twreporter/twreporter-redux/pull/22 first.

**Comment by YuCJ** [comment]
> Please merge this one #22 first.

👍 #22 was just merged

File: src/component/provider.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ReduxStoreContext from '../context/redux-store'
+
+class Provider extends Component {
+  constructor(props) {
+    super(props)
+    const { store } = this.props
+    this.state = store
```
Comment (nickhsine) [inline src/component/provider.js]: `this.store = store`  or `this.state = store.getState()` ?

Comment (YuCJ) [reply src/component/provider.js]: I've added a commit to fix this

Comment (YuCJ) [reply src/component/provider.js]: I've added a commit to fix this

**Comment by nickhsine** [comment]
LGTM

**Pull request merged**